src/cc/table_desc.h

```diff
@@ -93,10 +81,10 @@ class TableDesc {
         leaf_snprintf(nullptr),
         is_shared(false),
         is_extern(false) {}
-  TableDesc(const std::string &name, FileDesc &&fd, int type, size_t key_size,
+  TableDesc(const std::string &name, int fd, int type, size_t key_size,
```

Comment:
I didn't want to add a `TableDesc` as `friend class` to this generic common helper. Actually, the only [call site](https://github.com/iovisor/bcc/blob/master/src/cc/frontends/b/codegen_llvm.cc#L1245) of the constructor I can found passes `int` value on this parameter.

But you are right on potential double-closing. How about we make this still take a `FileDesc &&`, but explicitly construct a `FileDesc` instance to pass to constructor in the call site?